### PR TITLE
Feature/rename tee enclave provider

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

File: src/main/java/com/iexec/common/tee/TeeUtils.java [hunk 1]
```diff
@@ -99,6 +104,23 @@ public static TeeEnclaveProvider getTeeEnclaveProvider(String hexTag) {
         return null;
     }
 
+    /**
+     * Returns enclave provider matching given {@code hexTag}.
```
Comment (mcornaton) [inline src/main/java/com/iexec/common/tee/TeeUtils.java]:
```suggestion
     * Returns framework matching given {@code hexTag}.
```

**Comment by mcornaton** [comment]
Could you please add `@deprecated` JavaDoc tags to point to the new `TeeFramework` enum?

**Review by mcornaton — APPROVED** [review]
Thanks! :)

File: src/main/java/com/iexec/common/tee/TeeEnclaveConfiguration.java [hunk 2]
```diff
@@ -30,7 +31,11 @@
 @Builder
 public class TeeEnclaveConfiguration {
 
+    /** @deprecated */
+    @Deprecated(forRemoval = false)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
     private TeeEnclaveProvider provider;
+    private TeeFramework framework;
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/tee/TeeEnclaveConfiguration.java:38]:
```suggestion
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private TeeFramework framework;
```
?

Comment (jbern0rd) [reply src/main/java/com/iexec/common/tee/TeeEnclaveConfiguration.java:38]: I am not sure about this. `TeeFramework` should never be null here.
Unless I'm missing something, a `TeeEnclaveConfiguration` not backed on a TEE framework has no meaning.
I added the annotation on `provider` to cleanup the JSON when core and worker will shift from `provider` to `framework` field.

Comment (james-toussaint) [reply src/main/java/com/iexec/common/tee/TeeEnclaveConfiguration.java:38]: Your are right, I was think about `@NonNull` !

**Review by james-toussaint — APPROVED** [review]
Thanks :)

**Pull request merged**